hack/testing/e2e_deployment.sh

```diff
@@ -0,0 +1,162 @@
+    tmp=$(getopt \
+        --options '' \
+        --long "$long" \
+        --name "$(basename "$0")" -- "$@")
+    eval set -- "$tmp"
+    while [ "$1" != -- ]; do
+        case "$1" in
+            --bootstrap-os) BOOTSTRAP_OS=bootstrap_os; shift;;
+            --build-images) BUILD_IMAGES=build_images; shift;;
+            --local-image) LOCAL_IMAGE=local_image; shift;;
+            --deploy) DEPLOY=deploy; shift;;
+            --cleanup) CLEANUP=cleanup; shift;;
+            --tmp-project) TMP_PROJECT=tmp_project; shift;;
+        esac
+    done
+    if [ ! "$DEPLOY" ]; then
+        [ "$TMP_PROJECT" ] \
+            && args_error 'can only use temporary project when deploying'
+        [ "$CLEANUP" ] \
+            && args_error 'can only use "cleanup" when deploying'
+    fi
+    if [ "$TMP_PROJECT" ]; then
+        TMP_PROJECT=apiman-e2e-test-$( \
+            base64 /dev/urandom | tr -cd [a-z0-9] | head -c 5 || true)
+    else
+        [ "$CLEANUP" ] \
+            && args_error 'can only use "cleanup" with a temporary project'
+    fi
+    [ "$BUILD_IMAGES" -a "$LOCAL_IMAGE" ] \
+        && args_error 'can'\''t use "local image" and "build images" together'
+    return 0
+}
+
+args_error() {
+    printf >&2 "%s: %s\n" "$(basename "$0")" "$@"
+    exit 1
+}
+
+bootstrap_os() {
+    os::util::environment::use_sudo
+    os::util::environment::setup_all_server_vars origin-apiman
+    configure_os_server
+    start_os_server
+    install_registry
+    wait_for_registry
+    export KUBECONFIG="${ADMIN_KUBECONFIG}"
+}
+
+setup_for_deployer() {
+    local build_images=$1 filter
+    os::cmd::expect_success \
+        'oc create -f "$ORIGIN_APIMAN_ROOT/deployer/deployer.yaml"'
+    os::cmd::expect_success \
+        'oc process template/apiman-deployer-account-template | oc create -f -'
+    os::cmd::expect_success \
+        'oadm policy add-role-to-user edit --serviceaccount apiman-deployer'
+    os::cmd::expect_success \
+        "$(echo \
+            oadm policy add-cluster-role-to-user cluster-reader \
+            "system:serviceaccount:$(oc project --short):apiman-gateway")"
+    os::cmd::expect_success \
+        'oc secrets new apiman-deployer nothing=/dev/null'
+    os::cmd::expect_success \
+        'oc label secret/apiman-deployer apiman-infra=deployer'
+    [ "$build_images" ] \
+        && filter="jq '{apiVersion,kind,items:[.items[]|del(.spec.triggers)]}'"
+    os::cmd::expect_success \
+        "$(echo \
+            oc process -f '"$ORIGIN_APIMAN_ROOT/hack/dev-builds.yaml"' \
+            \| $filter \| oc create -f -)"
+}
+
+build_images() {
+    local get_is_tags
+    get_is_tags='{{range .status.tags}}{{.tag}}{{"\n"}}{{end}}'
+    get_is_tags="oc get --template '$get_is_tags' imagestream"
+    os::cmd::try_until_text \
+        "$get_is_tags origin" '^latest$' "$((2 * TIME_MIN ))" 10
+    os::cmd::try_until_text \
+        "$get_is_tags centos" '^7$' "$((2 * TIME_MIN ))" 10
+    for x in deployer elasticsearch curator; do
```

Comment:
we can add now: builder console gateway, except the last two need to build from a different repo... guess we can assume that's cloned beside this repo.